codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/CodegenTarget.kt

```diff
@@ -11,3 +11,21 @@ package software.amazon.smithy.rust.codegen.core.smithy
 enum class CodegenTarget {
     CLIENT, SERVER
 }
+
+/**
+ * Convenience extension to execute thunk if the target is for CodegenTarget.CLIENT
+ */
+fun <B> CodegenTarget.ifClient(thunk: () -> B): B? = if (this == CodegenTarget.CLIENT) {
```

Comment:
Nice! These are useful.

Nit: These can be implemented directly on the enum class, and then they won't need to be separately imported where they're used. IIRC, the syntax is:

```kotlin
enum class SomeEnum {
    SOMETHING, ANOTHER_THING;

    fun something() { /* ... */ }
}
```

Reply:
Thanks for the suggestion. Addressed in 7b719e1.